instfixlib.sh

```diff
@@ -0,0 +1,535 @@
+	# Activate the change
+	if [ "$IS_ROOT" = "true" ]; then
+		source "$NPM_COMMAND_FIX_PATH"
+	fi
+
+	# Make sure the bashrc file exists - it should, but you never know...
+	sudo touch /root/.bashrc
+	# If .bashrc does not contain the source command, we need to add it
+	sudo grep -q -E "^source /root/\.iobroker/npm_command_fix" /root/.bashrc &> /dev/null
+	if [ $? -ne 0 ]; then
+		append_to_file "$BASHRC_LINES" /root/.bashrc
+	fi
+}
+
+set_root_permissions() {
+	file="$1"
+	$SUDOX chown root:$ROOT_GROUP $file
+	$SUDOX chmod 755 $file
+}
+
+make_executable() {
+	file="$1"
+	$SUDOX chmod 755 $file
+}
+
+change_owner() {
+	user="$1"
+	file="$2"
+	if [ "$HOST_PLATFORM" == "osx" ]; then
+		owner="$user"
+	else
+		owner="$user:$user"
+	fi
+	cmdline="$SUDOX chown"
+	if [ -d $file ]; then
+		# recursively chown directories
+		cmdline="$cmdline -R"
+	elif [ -L $file ]; then
+		# change ownership of symbolic links
+		cmdline="$cmdline -h"
+	fi
+	$cmdline $owner $file
+}
+
+function add2sudoers() {
+	local xsudoers=$1
+	shift
+	xarry=("$@")
+	for cmd in "${xarry[@]}"; do
+		# Test each command if and where it is installed
+		cmd_bin=$(echo $cmd | cut -d ' ' -f1)
+		cmd_path=$(which $cmd_bin 2> /dev/null)
+		if [ $? -eq 0 ]; then
+			# Then add the command to SUDOERS_CONTENT
+			full_cmd=$(echo "$cmd" | sed -e "s|$cmd_bin|$cmd_path|")
+			SUDOERS_CONTENT+=$xsudoers"NOPASSWD: $full_cmd\n"
+		fi
+	done
+}
+
+create_user_linux() {
+	username="$1"
+	id "$username" &> /dev/null;
+	if [ $? -ne 0 ]; then
+		# User does not exist
+		$SUDOX useradd -m -s /usr/sbin/nologin "$username"
+		echo "User $username created"
+	fi
+	# Add the current non-root user to the iobroker group so he can access the iobroker dir
+	if [ "$username" != "$USER" ] && [ "$IS_ROOT" = false ]; then
+		sudo usermod -a -G $username $USER
+	fi
+
+	SUDOERS_CONTENT="$username ALL=(ALL) ALL\n"
+	# Add the user to all groups we need and give him passwordless sudo privileges
+	# Define which commands iobroker may execute as sudo without password
+	declare -a iob_commands=(
+		"shutdown -h now" "halt" "poweroff" "reboot"
+		"systemctl start" "systemctl stop"
+		"mount" "umount" "systemd-run"
+		"apt-get" "apt" "dpkg" "make"
+		"ping" "fping"
+		"arp-scan"
+		"setcap"
+		"vcgencmd"
+		"cat"
+		"df"
+		"mysqldump"
+		"ldconfig"
+	)
+	add2sudoers "$username ALL=(ALL) " "${iob_commands[@]}"
+
+	# Additionally, define which iobroker-related commands may be executed by every user
+	declare -a all_user_commands=(
+		"systemctl start iobroker"
+		"systemctl stop iobroker"
+		"systemctl restart iobroker"
+	)
+	add2sudoers "ALL ALL=" "${all_user_commands[@]}"
+
+	# Furthermore, allow all users to execute node iobroker.js as iobroker
+	if [ "$IOB_USER" != "$USER" ]; then
+		add2sudoers "ALL ALL=($IOB_USER) " "node $CONTROLLER_DIR/iobroker.js"
+	fi
+
+	SUDOERS_FILE="/etc/sudoers.d/iobroker"
+
+#ADOE: IF (INSTALLER) THEN ...
+	$SUDOX rm -f $SUDOERS_FILE
+#ADOE: ENDIF
```

Comment:
dito

Reply:
see above.

Reply:
In this case, the `rm` existed only in the INSTALLER. And i was not sure if also needed in the FIXER.

Reply:
I think it only existed in the fixer because the installer assumed the file would not be there. Lets put it in both.